rir/src/interpreter/interp.cpp

```diff
@@ -785,14 +1043,83 @@ RIR_INLINE SEXP rirCall(CallContext& call, Context* ctx) {
     return result;
 }
 
+#ifdef DEBUG_SLOWCASES
+
+class SlowcaseCounter {
+  public:
+    std::unordered_map<std::string, size_t> counter;
+
+    void count(const std::string& kind, CallContext& call, Context* ctx) {
+        std::stringstream message;
+        message << "Fast case " << kind << " failed for "
+                << getBuiltinName(getBuiltinNr(call.callee)) << " ("
+                << getBuiltinNr(call.callee) << ") "
+                << "nargs : " << call.suppliedArgs;
+        if (call.suppliedArgs > 0) {
+            auto arg = call.stackArg(0);
+            if (TYPEOF(arg) == PROMSXP)
+                arg = PRVALUE(arg);
+            if (arg == R_UnboundValue)
+                message << "arg0 lazy";
+            else if (arg == R_MissingArg)
+                message << "arg0 missing";
+            else
+                message << " arg0 : " << type2char(TYPEOF(arg)) << " a "
+                        << (ATTRIB(arg) != R_NilValue);
+        }
+        if (!counter.count(message.str()))
+            counter[message.str()] = 0;
+        counter[message.str()]++;
```

Comment:
I think I noticed this pattern somewhere else already..
The check if the key is present is not necessary. `operator[]` will insert a new default constructed value if the key is missing :)

Reply:
initial value of primitive types is undefined. You are not guaranteed to get `0`.

Reply:
Not true :) Map apparently does something called value initialization
https://stackoverflow.com/a/50371768/9977179

Reply:
f* me. C++

Reply:
always something unexpected to discover, thanks :)